### ISSUE TITLE
Profiler does not work on a virtualized ARM64 Mac

### ISSUE THREAD

**Issue by apangin** [jvm-profiling-tools/async-profiler#618]
SIGILL at `JitWriteProtection::JitWriteProtection(bool)+0x18`.
The address corresponds to `MRS` instruction, unsupported in a virtual machine.

```
Host: "VirtualMac2,1" arm64, 2 cores, 4G, Darwin 21.5.0, macOS 12.4 (21F79)
```

**Comment by apangin** [comment]
On the virtual macOS, `pthread_jit_write_protect_supported_np` returns false, so JitWriteProtection should not be used.

**Comment by apangin** [comment]
Fixed in 2.8.3
